tests/unittest/others/test_perf_sanity_bounds.py

```
@@ -0,0 +1,194 @@
+    assert proc.poll() is not None, "process should be reaped after SIGTERM"
+
+
+def test_stop_process_escalates_to_sigkill_when_sigterm_is_ignored():
+    """A rank wedged in native code never runs the SIGTERM handler.
+
+    The bare terminate(); wait() this replaces would block teardown forever.
+    """
+    ignores_sigterm = (
+        "import signal, time\nsignal.signal(signal.SIGTERM, signal.SIG_IGN)\ntime.sleep(60)\n"
+    )
+    proc = subprocess.Popen([sys.executable, "-c", ignores_sigterm])
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)